runner/tests/operator/argos_operator/v2_2_0/test_retrieve_samples_by_query.py

```diff
@@ -0,0 +1,510 @@
+            file_name="FROZENPOOLEDNORMAL.R2.fastq",
+            path="/FROZENPOOLEDNORMAL.R2.fastq",
+        )
+        FileMetadata.objects.create_or_update(
+            file=poolednormal_R2_file_instance,
+            metadata={},
+        )
+
+        poolednormal_R1_file_instance_earliest = File.objects.create(
+            file_type=fastq_filetype_instance,
+            file_group=poolednormal_filegroup_instance,
+            file_name="FROZENPOOLEDNORMAL.CORRECT.R1.fastq",
+            path="/FROZENPOOLEDNORMAL.CORRECT.R1.fastq",
+        )
+        FileMetadata.objects.create_or_update(
+            file=poolednormal_R1_file_instance_earliest,
+            metadata={},
+        )
+        poolednormal_R2_file_instance_earliest = File.objects.create(
+            file_type=fastq_filetype_instance,
+            file_group=poolednormal_filegroup_instance,
+            file_name="FROZENPOOLEDNORMAL.CORRECT.R2.fastq",
+            path="/FROZENPOOLEDNORMAL.CORRECT.R2.fastq",
+        )
+        FileMetadata.objects.create_or_update(
+            file=poolednormal_R2_file_instance_earliest,
+            metadata={},
+        )
+
+        pooled_normals = get_pooled_normals(
+            run_ids=["FAUCI2_0049", "BONO_12314"], preservation_types=["Frozen"], bait_set="IMPACT505_BAITS"
+        )
+        # remove the R1_bid and R2_bid for testing because they are non-deterministic
+        # TODO: mock this ^^
+        pooled_normals["R1_bid"].pop()
+        pooled_normals["R2_bid"].pop()
+
+        expected_pooled_normals = {
+            "CN": "MSKCC",
+            "PL": "Illumina",
+            "PU": ["PN_FCID_FROZENPOOLEDNORMAL"],
+            "LB": "IMPACT505_BAITS_FROZEN_BONO_POOLEDNORMAL_1",
+            "tumor_type": "Normal",
+            "ID": ["IMPACT505_BAITS_FROZEN_BONO_POOLEDNORMAL_PN_FCID_FROZENPOOLEDNORMAL"],
+            "SM": "IMPACT505_BAITS_FROZEN_BONO_POOLEDNORMAL",
+            "species": "",
+            "patient_id": "PN_PATIENT_ID",
+            "bait_set": "impact505_baits",
+            "sample_id": "IMPACT505_BAITS_FROZEN_BONO_POOLEDNORMAL",
+            "run_date": [""],
+            "specimen_type": "Pooled Normal",
+            "R1": ["/FROZENPOOLEDNORMAL.CORRECT.R1.fastq"],
+            "R2": ["/FROZENPOOLEDNORMAL.CORRECT.R2.fastq"],
+            "R1_bid": [],
+            "R2_bid": [],
+            "bam": [],
+            "bam_bid": [],
+            "request_id": "IMPACT505_BAITS_FROZEN_BONO_POOLEDNORMAL",
+            "pi": "",
+            "pi_email": "",
+            "run_id": ["FAUCI2_0049", "BONO_12314"],
+            "preservation_type": [["Frozen"]],
+            "run_mode": "",
+        }
+
+        pooled_normals_sorted = {k: sorted(v) for k, v in pooled_normals.items()}
+        expected_pooled_normals_sorted = {k: sorted(v) for k, v in expected_pooled_normals.items()}
+
+        self.assertEqual(pooled_normals_sorted, expected_pooled_normals_sorted)
```

Comment:
This is the tests. First test retrieves one set of pooled normals; the other performs the tiebreaker, returning only the _**earliest pooled normal**_ when the tumor has multiple matches.